core/templates/dev/head/filters.js

```diff
@@ -196,8 +196,28 @@ oppia.filter('parameterizeRuleDescription', [
           } else {
             // The following case is for MultipleChoiceInput
             for (var i = 0; i < choices.length; i++) {
+              var stringChange = true;
+              var choicesLabelString = choices[i].label;
+              while(stringChange) {
+                var nonInteractiveInputs = ['collapsible', 'image', 'link',
+                                            'math', 'tabs', 'video'];
+                for (var j = 0; j < nonInteractiveInputs.length; j++) {
+                  if (choicesLabelString.includes('<oppia-noninteractive-' +
+                    nonInteractiveInputs[j])) {
+                    var testString = '<oppia-noninteractive-' +
+                    nonInteractiveInputs[j];
+                    var regex = new RegExp('(' + testString + '([^>]+)>)');
+                    choicesLabelString = choicesLabelString.replace(regex, '[' +
+                    nonInteractiveInputs[j] + ']');
+                    stringChange = true;
+                    break;
```

Comment:
What happens if there are multiple images, etc in a choice?

Reply:
The outer while loop will run till there is no NonInteractiveInput opening tag is present in the string. And in every iteration, the code replaces the opening tag with [NonInteractiveInput]. Then the filter 'convertToPlainText' takes care of the rest. Hence, this takes care of multiple images, links, text etc.